python/app/dialog.py

```diff
@@ -978,13 +1022,30 @@ def build_entity_tabs(self):
             # view (QListView), filter (QCheckbox)
             if data["has_description"]:
                 label = self.create_entity_tab_label(entity_tab_name, tab_widget)
-                tab_widget.layout().addWidget(label)
+                if data["entity_type"] == "Task":
+                    # Create the Label Vertical Layout
+                    label_lay = QtGui.QHBoxLayout()
+                    label_lay.addWidget(label)
+                    # Nest the label layout to the label_menus layout
+                    self._label_menus_layout.addLayout(label_lay)
+                else:
+                    tab_widget.layout().addWidget(label)
                 data["description"] = label
 
             if data["has_view"]:
                 view = self.create_entity_tab_view(entity_tab_name, tab_widget)
-                tab_widget.layout().addWidget(view)
-                data["view"] = view
+                if data["entity_type"] == "Task":
+                    data["view"] = view
+                    data["widget"] = tab_widget
+                    # If entity_type is "Task", run setup_task_menus() before adding the view
+                    # to the widget to place the sort and filter menus at the top right of My tasks tab
+                    entity_data = self.setup_task_menus(data)
+                    # Add the view to the widget
+                    entity_data["widget"].layout().addWidget(view)
```

Comment:
Same thing here, this should be generic to add filter/sort menu for any entity type